DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -304,13 +304,22 @@ public function applyDefaults(ContainerBuilder $container, $serviceId, array $at
 
         $definition->addMethodCall('setLabel', array($label));
 
+
+        $persistFilters = $container->getParameter('sonata.admin.configuration.filters.persist');
+        // override default configuration with admin config if set
         if (isset($attributes['persist_filters'])) {
-            $persistFilters = (bool) $attributes['persist_filters'];
-        } else {
-            $persistFilters = (bool) $container->getParameter('sonata.admin.configuration.filters.persist');
+            $persistFilters = $attributes['persist_filters'];
         }
 
-        $definition->addMethodCall('setPersistFilters', array($persistFilters));
+        // if boolean config value provided, use the default persister : session
+        if (is_bool($persistFilters) && $persistFilters) {
```

Comment:
you got it. configuring `persist_filters` to `true` means "i don't care how you persist it, but persist it" :)